### PR TITLE
feat(cli): files download respects filename path

### PR DESCRIPTION
**Pull request by iancoleman**
The download location is no longer $HOME/Downloads, instead using $PWD to store the file relative to where the download command is run. This can be used with both relative paths and absolute paths. This was discussed by the community in [NewYearNewNet post 405](https://safenetforum.org/t/newyearnewnet-04-01-2024-testnet-offline/39034/455).

eg
`safe files download myfile.txt ADDR`
Destination before this PR: $HOME/download/myfile.txt
Destination after this PR: $PWD/myfile.txt

`safe files download my/relative/dir/myfile.txt ADDR`
Destination before this PR: $HOME/download/my/relative/dir/myfile.txt
Destination after this PR: $PWD/my/relative/dir/myfile.txt

`safe files download /tmp/myfile.txt ADDR`
Destination before this PR: $HOME/download/tmp/myfile.txt
Destination after this PR: /tmp/myfile.txt

## Description

reviewpad:summary

### CODE REVIEW

File: sn_cli/src/subcommands/files.rs [hunk 1]
```diff
@@ -137,11 +137,49 @@ pub(crate) async fn files_cmds(
                 );
             }
 
-            let download_dir = dirs_next::download_dir().unwrap_or(root_dir.to_path_buf());
+            let mut download_dir = root_dir.to_path_buf();
+            let mut download_file_name = file_name.clone();
+            if file_name.is_some() {
+                // file_name may direct the downloaded data to:
```
Comment (grumbach) [inline sn_cli/src/subcommands/files.rs]:
```suggestion
            if let Some(file_name) = file_name {
                // file_name may direct the downloaded data to:
```
With this we don't need the `.expect("...")` below

File: sn_cli/src/subcommands/files.rs [hunk 2]
```diff
@@ -137,11 +137,49 @@ pub(crate) async fn files_cmds(
...
+                //
+                // the current directory (just a filename)
+                // eg safe files download myfile.txt ADDRESS
+                //
+                // a directory relative to the current directory (relative filename)
+                // eg safe files download my/relative/path/myfile.txt ADDRESS
+                //
+                // a directory relative to root of the filesystem (absolute filename)
+                // eg safe files download /home/me/mydir/myfile.txt ADDRESS
+                let file_name_val = file_name.expect("file_name unspecified");
```
Comment (grumbach) [inline sn_cli/src/subcommands/files.rs]: This expect can be removed

File: sn_cli/src/subcommands/files.rs [hunk 3]
```diff
@@ -137,11 +137,49 @@ pub(crate) async fn files_cmds(
...
+                }
+                let file_name_dir = file_name_path.parent();
+                if file_name_dir.is_none() {
+                    // just a filename, use the current_dir
+                    download_dir = std::env::current_dir().unwrap_or(root_dir.to_path_buf());
+                } else if file_name_path.is_relative() {
+                    // relative to the current directory. Make the relative path
+                    // into an absolute path by joining it to current_dir
+                    let relative_dir =
+                        file_name_dir.expect("file_name relative directory unspecified");
```
Comment (grumbach) [inline sn_cli/src/subcommands/files.rs]: Same here, this expect could be removed with an `if let Some(file_name_dir) = file_name_dir` clause

**Review by grumbach — CHANGES_REQUESTED** [review]
Excellent suggestion!
Some tiny code improvement suggestions related to removable `.expect(...)` statements

**Comment by iancoleman** [comment]
Thanks @grumbach  for the tips with rust style, I really appreciate learning how to improve this sort of thing.

The suggestions were implemented as of commit https://github.com/maidsafe/safe_network/commit/e86be0f459e0111e605fda22d0c1dba9b8df446e

**Review by grumbach — APPROVED** [review]
LGTM

**Pull request merged**